### ISSUE TITLE
pywavelets implementation?

### ISSUE THREAD

**Issue by neosr-project** [JPWang-CS/FreqFormer#2]
Hi, thanks for your research. I see FreqFormer uses `pytorch_wavelets` for wavelet calculation, which hasn't been maintained for at least 2 years now. Do you know if it would be possible to replace the transform using [pywavelets](https://pypi.org/project/PyWavelets/) instead? According to the [DTCWT documentation](https://pytorch-wavelets.readthedocs.io/en/latest/dtcwt.html) it is possible to replicate it using `morlet` or `gabor`, which pywt [seems to support](https://pywavelets.readthedocs.io/en/latest/ref/cwt.html#wavelet-bandwidth-and-center-frequencies). However, I don't have enough knowledge in wavelet transforms to show equivalence of both methods. Any help is appreciated.

**Comment by JPWang-CS** [comment]
> Hi, thanks for your research. I see FreqFormer uses `pytorch_wavelets` for wavelet calculation, which hasn't been maintained for at least 2 years now. Do you know if it would be possible to replace the transform using [pywavelets](https://pypi.org/project/PyWavelets/) instead? According to the [DTCWT documentation](https://pytorch-wavelets.readthedocs.io/en/latest/dtcwt.html) it is possible to replicate it using `morlet` or `gabor`, which pywt [seems to support](https://pywavelets.readthedocs.io/en/latest/ref/cwt.html#wavelet-bandwidth-and-center-frequencies). However, I don't have enough knowledge in wavelet transforms to show equivalence of both methods. Any help is appreciated.

The main reason for using this library is that it supports tensor input and output. I haven't conducted experiments on whether the conversion between np types and tensors affects gradient backpropagation, but you can try training to see if it works.
